backendv2/pkg/module/indexer.go

```diff
@@ -0,0 +1,869 @@
+}
+
+// collectSubmodulesData collects data from all submodules
+func (m *Reader) collectSubmodulesData(ctx context.Context, workDir string) (map[string]map[string]interface{}, error) {
+	ctx, span := m.tracer.Start(ctx, "module.collectSubmodulesData")
+	defer span.End()
+
+	submodulesData := make(map[string]map[string]interface{})
+	modulesDir := filepath.Join(workDir, "modules")
+
+	// Check if modules directory exists
+	if _, err := os.Stat(modulesDir); os.IsNotExist(err) {
+		return submodulesData, nil // No submodules
+	}
+
+	// Read the modules directory
+	entries, err := os.ReadDir(modulesDir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read modules directory: %w", err)
+	}
+
+	for _, entry := range entries {
+		if !entry.IsDir() {
+			continue
+		}
+
+		submoduleName := entry.Name()
+		fullSubmodulePath := filepath.Join(workDir, "modules", submoduleName)
+
+		// Run tofu show on the submodule
+		tofuJSON, err := m.runTofuShow(ctx, fullSubmodulePath)
+		if err != nil {
+			slog.WarnContext(ctx, "Failed to run tofu show on submodule",
+				"submodule", submoduleName, "error", err)
+			continue // Don't fail the entire process for one submodule
+		}
+
+		submodulesData[submoduleName] = tofuJSON
+	}
+
+	return submodulesData, nil
+}
+
+// collectExamplesData collects data from all examples
+func (m *Reader) collectExamplesData(ctx context.Context, workDir string) (map[string]map[string]interface{}, error) {
+	ctx, span := m.tracer.Start(ctx, "module.collectExamplesData")
+	defer span.End()
+
+	examplesData := make(map[string]map[string]interface{})
+	examplesDir := filepath.Join(workDir, "examples")
+
+	// Check if examples directory exists
+	if _, err := os.Stat(examplesDir); os.IsNotExist(err) {
+		return examplesData, nil // No examples
+	}
+
+	// Read the examples directory
+	entries, err := os.ReadDir(examplesDir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read examples directory: %w", err)
+	}
+
+	for _, entry := range entries {
+		if !entry.IsDir() {
+			continue
+		}
+
+		exampleName := entry.Name()
+		fullExamplePath := filepath.Join(workDir, "examples", exampleName)
+
+		// Run tofu show on the example
+		tofuJSON, err := m.runTofuShow(ctx, fullExamplePath)
+		if err != nil {
+			slog.WarnContext(ctx, "Failed to run tofu show on example",
+				"example", exampleName, "error", err)
+			continue // Don't fail the entire process for one example
+		}
+
+		examplesData[exampleName] = tofuJSON
+	}
+
+	return examplesData, nil
+}
+
+// readAndParseModule reads and parses module files from the working directory
+func (m *Reader) readAndParseModule(ctx context.Context, workDir string) (map[string]interface{}, error) {
+	ctx, span := m.tracer.Start(ctx, "module.readAndParseModule")
+	defer span.End()
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		span.RecordError(err)
+		return nil, fmt.Errorf("failed to get current working directory: %w", err)
+	}
+	stat, err := os.Stat(path.Join(cwd, "tofu"))
```

Comment:
This feels like it should determined when creating the Reader and stored as m.TofuExecPath